### PR TITLE
Property tests

### PR DESCRIPTION
**Pull request by henrikrudstrom**
Added more tests:
- test basic property types
- test aliased property
- test bindings immediate update
- test reference (grand)parents by id

the last test tries to reference a parent by its Id, but it fails with an error on the parser level.

### CODE REVIEW

**Comment by ChALkeR** [comment]
Great, thanks!

This looks good to me, except minor formatting nits.
A quick question: does `*.qml` files have inconsistent spacing on a purpose?

**Comment by henrikrudstrom** [comment]
you mean 4 space tabs?

**Comment by ChALkeR** [comment]
`Item {` / `Item{` inconsistency, some extra newlines, missing newlines at the end of file, etc.

File: tests/QMLEngine/javascript.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Created by henrik on 21.02.16.
```
Comment (ChALkeR) [inline tests/QMLEngine/javascript.js]: Is this needed? This would be visible in `git blame` and `git log`.

File: tests/QMLEngine/javascript.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ */
+describe('QMLEngine.javascript', function() {
+    var loader = prefixedQmlLoader('QMLEngine/qml/Javascript');
+    it('can be parsed', function() {
+        var div = loader('BasicSyntax');
+        div.remove()
+
+    });
+});
```
Comment (ChALkeR) [inline tests/QMLEngine/javascript.js]: Missing newline at the end of file.

File: tests/QMLEngine/javascript.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Created by henrik on 21.02.16.
+ */
+describe('QMLEngine.javascript', function() {
+    var loader = prefixedQmlLoader('QMLEngine/qml/Javascript');
+    it('can be parsed', function() {
+        var div = loader('BasicSyntax');
+        div.remove()
+
```
Comment (ChALkeR) [inline tests/QMLEngine/javascript.js]: Extra newline

File: tests/QMLEngine/qml/PropertiesBasic.qml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Item{
```
Comment (ChALkeR) [inline tests/QMLEngine/qml/PropertiesBasic.qml]: If this is not to test parser behaviour here, could you use `Item {`, please?
If this is for parser tests — could you create a separate entry for them?

**Comment by ChALkeR** [comment]
@henrikrudstrom I just filed #64 that enables eslint for `js` files in the `./tests` folder. Take a look, please.

**Comment by henrikrudstrom** [comment]
was just about to ask about eslint... you nasty nitpicker ;)

**Comment by henrikrudstrom** [comment]
fixed the qml and some newlines, lets make eslint do the rest

**Comment by henrikrudstrom** [comment]
rebased it, ok to merge now?

**Comment by ChALkeR** [comment]
@henrikrudstrom There are still whitespace issues, but I will fix them when landing.
Everything else LGTM, merging.

**Comment by ChALkeR** [comment]
@henrikrudstrom I squashed some of your changes (i.e. moved them between commits) and performed minor code formatting changes.

There also was an minor issue that I fixed in a separate commit — you forgot to remove the html elements =).

**Pull request merged**